### DISCUSSION TITLE
How should I create my String variables? with class or Not?

### DISCUSSION THREAD

**Discussion by harsh4248** [community/community#20150] [Mobile] [answered]
I'm new on FLutter and I'm working on Flutter app projects for more learn and I have a question about the keeping String values. On the Some flutter app tutorial videos, the guy keeping all String value in any dart file. For example:

```
//Login Page
const String loginButton = 'LogIn';
const String registerQuestionText = 'Do You Have Any Account?';
const String registerButtonText = 'SingIn!';
const String usernameText = 'Username';
const String passwordText = 'Password';


//Register Page
const String emailText = 'Email';
const String bookText = 'Favorite Book';
const String passwordVerifyText = 'Password Again';
```
here on the other way, someones doing like that:

```
class ProjectKeys {
  static final String appBarTitle = "Hello";
  static final String buttonLabel = "Press The Button";
}
```

So, I'm confused about these usages. Which one should I use for clean code? Thanks in advance..

**Comment by ketan-rathod-713** [comment]
If your string values are only used in that page only then I think keeping it local is more convenient as using a separate class for it and then importing it may create more confusions ,
 
but if your strings are used in multiple screens then you can go for 2nd approach .

**Comment by chiragmehta900** [comment, marked as answer]
You can use it like this. 

```
class ProjectKeys {
  static final String appBarTitle = "Hello";
  static final String buttonLabel = "Press The Button";
}
```

if you're using this you can use your String as ProjectKeys.appBarTitle
which is object-based approach
